src/Database.js

```diff
@@ -38,21 +38,31 @@ class Database {
     });
   }
 
-  run(sql, ...params) {
-    return new this.Promise((resolve, reject) => {
-      this.driver.run(sql, params || [], function runExecResult(err) {
+  run(sql) {
+    const params = (arguments.length === 2 && typeof arguments[1] === 'object'
```

Comment:
this logic seems to be spread through most methods, should be its own private method?